scripts/docgen/snippets/writeFilesConsumer.ts

```diff
@@ -0,0 +1,27 @@
+import { writeFileSync } from "node:fs"
+import { join } from "node:path"
+import { DocGenSnippetConsumer } from "../config.js"
+
+export type CreateWriteFilesConsumerOptions = {
+    rootOutDir: string
+    transformFileName?: (name: string) => string
+}
+
+export const createWriteFilesConsumer =
+    (options: CreateWriteFilesConsumerOptions): DocGenSnippetConsumer =>
+    (snippets) => {
+        const obj = Object.fromEntries(
```

Comment:
I might have randomly named this, also probably my bad. Let's call it something clearer like `snippetsByRelativeOutPath`.